iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/posix_call.inl

```diff
@@ -73,9 +73,7 @@ inline cxx::string<POSIX_CALL_ERROR_STRING_SIZE> errorLiteralToString(const char
 template <typename T>
 inline cxx::string<POSIX_CALL_ERROR_STRING_SIZE> PosixCallResult<T>::getHumanReadableErrnum() const noexcept
 {
-    /// NOLINTJUSTIFICATION @todo iox-#1614 replace with upcoming uninitialized array
-    /// NOLINTNEXTLINE(hicpp-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    char buffer[POSIX_CALL_ERROR_STRING_SIZE];
+    containers::UninitializedArray<char, POSIX_CALL_ERROR_STRING_SIZE, containers::ZeroedBuffer> buffer;
```

Comment:
Zeroing the whole buffer should be avoided. Can we get around this? These are essentially memsets that we are accumulating and can avoid.

I think we only should be set the zero terminator afterwards in e.g. `errorLiteralToString` which already needs to deal with the buffer being to small (by truncation, I assume).

Reply:
I reverted the changes and adapted the justification. It makes sense to use a char array since this required by the POSIX function. Even though this can be circumvented, at this point we are already in a POSIX call wrapper.